src/core/ApiStore.ts

```diff
@@ -328,48 +328,76 @@ export async function fetchUser(id: string): Promise<DefaultApiResponse<UserInfo
 	}
 }
 
+function setFinalReports(final_evaluations: FinalEvaluation[], reports: ReportCard[]): void {
+	for (const ev of final_evaluations) {
+		let status = ReportApprovalState.NA;
+
+		switch (ev.status) {
+			case "pending":
+				status = ReportApprovalState.Pending;
+				break;
+			case "passed":
+				status = ReportApprovalState.Approved;
+				break;
+			case "failed":
+				status = ReportApprovalState.Failed;
+				break;
+		}
+
+		const newReport: ReportCard = {
+			id: ev.id,
+			group: ev.group ? ev.group.grade_name : "",
+			starting_month: new Date(),
+			ending_month: new Date(),
+			year: ev.group ? ev.group.year : "",
+			type: "Final",
+			passed: status,
+			report_url: ev.report_card_url,
+		};
+		reports.push(newReport);
+	}
+}
+
+function setIntermediateReports(intermediate_evaluations: IntermediateEvaluation[], reports: ReportCard[]): void {
+	for (const ev of intermediate_evaluations) {
+		const start_month = ev.starting_month.replaceAll("-", "/");
+		const end_month = ev.ending_month.replaceAll("-", "/");
+
+		const newReport: ReportCard = {
+			id: ev.id,
+			group: ev.group ? ev.group.grade_name : "",
+			starting_month: new Date(start_month),
+			ending_month: new Date(end_month),
+			year: "",
+			type: "Intermedio",
+			passed: ReportApprovalState.NA,
+			report_url: ev.report_card_url,
+		};
+		reports.push(newReport);
+	}
+}
+
```

Comment:
These shouldn´t be here. Try with the core helper. And also instead of mutating the reports param, build an array and return it.

Reply:
Ready.